File: apps/mobile/ios/Visvine/Core/AppConfig.swift
import Foundation

/// Build/runtime configuration, sourced from Info.plist keys. Replaces the old
/// `EXPO_PUBLIC_*` values that were inlined into the JS bundle.
enum AppConfig {
    static let deepLinkScheme = "visvine"

    /// Backend origin (no trailing slash), e.g. http://localhost:3000.
    static let apiBase: String = {
        let raw = (infoString("VisvineApiBaseURL")) ?? "http://localhost:3000"
        return raw.hasSuffix("/") ? String(raw.dropLast()) : raw
    }()

    /// Whether the in-app "Dev login" path is offered.
    static let devAuthEnabled: Bool = {
        if let b = Bundle.main.object(forInfoDictionaryKey: "VisvineDevAuthEnabled") as? Bool { return b }
        return infoString("VisvineDevAuthEnabled") == "true"
    }()

    /// OAuth web client id; required only for real Google sign-in.
    static let googleClientID: String = infoString("VisvineGoogleClientID") ?? ""

    private static func infoString(_ key: String) -> String? {
        guard let value = Bundle.main.object(forInfoDictionaryKey: key) as? String,
              !value.isEmpty else { return nil }
        return value
    }
}

File: apps/mobile/android/app/src/main/java/com/visvine/mobile/core/AppConfig.kt
package com.visvine.mobile.core

import com.visvine.mobile.BuildConfig

/**
 * Build-time configuration, sourced from BuildConfig fields wired in
 * app/build.gradle.kts (which in turn read gradle properties). Replaces the
 * old EXPO_PUBLIC_* environment values that were inlined into the JS bundle.
 */
object AppConfig {
    /** Backend origin, e.g. http://10.0.2.2:3000 (emulator → host). */
    val apiBaseUrl: String = BuildConfig.API_BASE_URL.trimEnd('/')

    /** Whether the in-app "Dev login" path is offered. */
    val devAuthEnabled: Boolean = BuildConfig.DEV_AUTH_ENABLED

    /** OAuth web client id; required only for real Google sign-in. */
    val googleClientId: String = BuildConfig.GOOGLE_CLIENT_ID

    /** Custom scheme used by the OAuth deep-link callback. */
    const val DEEP_LINK_SCHEME = "visvine"
}
